webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -220,8 +239,8 @@ public void writeStatusAndHeaders(Http.ResponseStatus status, Map<String, List<S
 
         // Content length optimization attempt
         if (!lengthOptimization) {
-            LOGGER.fine(() -> log("Writing headers %s", status));
             requestEntityAnalyzed = requestEntityAnalyzed.thenApply(listener -> {
+                LOGGER.fine(() -> log("Writing headers %s", status));
```

Comment:
Given the assignment in the line above, shouldn't we renamed `requestEntityAnalyzed`? Its name is too close to the new one added, yet its completion reflects more than just the request entity being processed?

Reply:
Not sure I understand - it is the same future. We just need to keep the original instance and complete on it, as the "last" isntance in the field will not reflect `thenApply` and other methods registered on previous instances.

Reply:
That's exactly the point, the last instance on this field is related to the writing of the response and no longer the analysis of the request entity (as its name suggests). My comment is just about naming, not semantics. Certainly not a stopper.